models/alexnet/alexnet_impl.hpp

```diff
@@ -0,0 +1,126 @@
+/**
+ * @file alexnet_impl.hpp
+ * @author Shubham Agrawal
+ *
+ * Implementation of AlexNet using mlpack.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MODELS_MODELS_ALEXNET_ALEXNET_IMPL_HPP
+#define MODELS_MODELS_ALEXNET_ALEXNET_IMPL_HPP
+
+#include "alexnet.hpp"
+
+namespace mlpack {
+namespace models {
+
+template<typename MatType>
+AlexNetType<MatType>::AlexNetType() :
+    ann::MultiLayer<MatType>(),
+    numClasses(1000),
+    includeTop(true)
+{
+  MakeModel();
+}
+
+template<typename MatType>
+AlexNetType<MatType>::AlexNetType(
+    const size_t numClasses,
+    const bool includeTop) :
+    ann::MultiLayer<MatType>(),
+    numClasses(numClasses),
+    includeTop(includeTop)
+{
+  MakeModel();
+}
```

Comment:
I think it's possible to use defaults to combine these two constructors, and then if you wanted you could inline `MakeModel()` directly into the constructor for simplicity, since I think it isn't used anywhere else.